File: platform/packages_mjs/env-resolver/src/env-aws-s3.x.ts
import { resolve, resolveBool, resolveInt } from '@internal/env-resolve';

export interface AwsS3Env {
    bucket: string | undefined;
    region: string;
    keyPrefix: string;
    ttl: string | undefined;
    accessKey: string | undefined;
    secretKey: string | undefined;
    endpoint: string | undefined;
    forcePathStyle: boolean;
    proxy: string | undefined;
    connectTimeout: number;
    readTimeout: number;
    maxRetries: number;
    verifySsl: boolean;
    debug: boolean;
}

export function resolveAwsS3Env(config?: Record<string, any>): AwsS3Env {
    return {
        bucket: resolve(undefined, ['AWS_S3_BUCKET', 'AWS_S3_BUCKETNAME'], config, 'bucket', undefined),
        region: resolve(undefined, ['AWS_S3_REGION', 'AWS_REGION', 'AWS_DEFAULT_REGION'], config, 'region', 'us-east-1'),
        keyPrefix: resolve(undefined, ['AWS_S3_KEY_PREFIX'], config, 'keyPrefix', 'jss3:'),
        ttl: resolve(undefined, ['AWS_S3_TTL'], config, 'ttl', undefined),
        accessKey: resolve(undefined, ['AWS_S3_ACCESS_KEY', 'AWS_ACCESS_KEY_ID'], config, 'accessKey', undefined),
        secretKey: resolve(undefined, ['AWS_S3_SECRET_KEY', 'AWS_SECRET_ACCESS_KEY'], config, 'secretKey', undefined),
        endpoint: resolve(undefined, ['AWS_S3_ENDPOINT', 'AWS_ENDPOINT_URL'], config, 'endpoint', undefined),
        forcePathStyle: resolveBool(undefined, ['AWS_S3_FORCE_PATH_STYLE'], config, 'forcePathStyle', false),
        proxy: resolve(undefined, ['AWS_S3_PROXY', 'HTTPS_PROXY'], config, 'proxy', undefined),
        connectTimeout: resolveInt(undefined, ['AWS_S3_CONNECT_TIMEOUT'], config, 'connectTimeout', 10),
        readTimeout: resolveInt(undefined, ['AWS_S3_READ_TIMEOUT'], config, 'readTimeout', 60),
        maxRetries: resolveInt(undefined, ['AWS_S3_MAX_RETRIES'], config, 'maxRetries', 3),
        verifySsl: resolveBool(undefined, ['AWS_S3_VERIFY_SSL'], config, 'verifySsl', false),
        debug: resolveBool(undefined, ['AWS_S3_DEBUG'], config, 'debug', false),
    };
}

File: platform/packages_py/env_resolver/env_resolver/env_aws_s3_x.py
from dataclasses import dataclass
from typing import Any

from env_resolve import resolve, resolve_bool, resolve_int


@dataclass(frozen=True)
class AwsS3Env:
    bucket: str | None
    region: str
    key_prefix: str
    ttl: str | None
    access_key: str | None
    secret_key: str | None
    endpoint: str | None
    force_path_style: bool
    proxy: str | None
    connect_timeout: int
    read_timeout: int
    max_retries: int
    verify_ssl: bool
    debug: bool


def resolve_aws_s3_env(config: dict[str, Any] | None = None) -> AwsS3Env:
    return AwsS3Env(
        bucket=resolve(None, ['AWS_S3_BUCKET', 'AWS_S3_BUCKETNAME'], config, 'bucket', None),
        region=resolve(None, ['AWS_S3_REGION', 'AWS_REGION', 'AWS_DEFAULT_REGION'], config, 'region', 'us-east-1'),
        key_prefix=resolve(None, ['AWS_S3_KEY_PREFIX'], config, 'key_prefix', 'jss3:'),
        ttl=resolve(None, ['AWS_S3_TTL'], config, 'ttl', None),
        access_key=resolve(None, ['AWS_S3_ACCESS_KEY', 'AWS_ACCESS_KEY_ID'], config, 'access_key', None),
        secret_key=resolve(None, ['AWS_S3_SECRET_KEY', 'AWS_SECRET_ACCESS_KEY'], config, 'secret_key', None),
        endpoint=resolve(None, ['AWS_S3_ENDPOINT', 'AWS_ENDPOINT_URL'], config, 'endpoint', None),
        force_path_style=resolve_bool(None, ['AWS_S3_FORCE_PATH_STYLE'], config, 'force_path_style', False),
        proxy=resolve(None, ['AWS_S3_PROXY', 'HTTPS_PROXY'], config, 'proxy', None),
        connect_timeout=resolve_int(None, ['AWS_S3_CONNECT_TIMEOUT'], config, 'connect_timeout', 10),
        read_timeout=resolve_int(None, ['AWS_S3_READ_TIMEOUT'], config, 'read_timeout', 60),
        max_retries=resolve_int(None, ['AWS_S3_MAX_RETRIES'], config, 'max_retries', 3),
        verify_ssl=resolve_bool(None, ['AWS_S3_VERIFY_SSL'], config, 'verify_ssl', False),
        debug=resolve_bool(None, ['AWS_S3_DEBUG'], config, 'debug', False),
    )
